server/test/controllers/device/device.controller.test.js

```diff
@@ -109,6 +109,34 @@ describe('GET /api/v1/device_feature/aggregated_states', () => {
   });
 });
 
+describe('GET /api/v1/device_feature/:device_feature_selector/states', () => {
+  beforeEach(async function BeforeEach() {
+    this.timeout(1000);
+    await insertStates(1);
+  });
+  it('should get device feature states by selector', async () => {
+    const now = new Date();
+    const dateState = `${now.getUTCFullYear()}-${`0${now.getUTCMonth() + 1}`.slice(-2)}-${`0${now.getUTCDate()}`.slice(
+      -2,
+    )}`;
+
+    await authenticatedRequest
+      .get('/api/v1/device_feature/test-device-feature/states')
+      .query({
+        from: new Date(`${dateState}T00:00:00.000Z`),
+        to: new Date(`${dateState}T23:59:59.999Z`),
+      })
+      .expect('Content-Type', /json/)
+      .expect(200)
+      .then((res) => {
+        expect(res.body).to.have.lengthOf(2000);
```

Comment:
It's a bit heavy to test returning 2000 state. The goal of inserting 2000 states for aggregated tests was to verify that only 100 of the 2000 were returned in the JSON response.

Here, testing that 10, 200, or 2000 are returned doesn't change anything, it'll just slow the test down on every CI run.

I would suggest testing only with like 50-100 states, it should be enough !